cpg-infrastructure/developer.rst

```diff
@@ -0,0 +1,142 @@
+CPG Infrastructure Developer Guide
+===================================
+
+Our CPG Infrastructure is our way of defining resources to make our analysis possible. This guide is intended for developers who want to contribute to the CPG Infrastructure. See the :doc:`user` for information on how to use the CPG Infrastructure.
+
+Repository structure
+--------------------
+
+The core of our infrastructure is defined in `cpg-infrastructure <https://github.com/populationgenomics/cpg-infrastructure>`_, and is made up of 3 levels.
+
+1. ``CPGInfrastructure``:
+    - Infrastructure defined ONCE CPG wide, so all datasets can reuse them.
+    - Manages group membership, see below for more details about that.
+    - Coordinates the level below.
+2. ``CPGDatasetInfrastructure``
+    - Infrastructure defined ONCE for a specific dataset.
+    - Pretty light, mostly the metamist project.
+3. ``CPGDatasetCloudInfrastructure```
+    - All the resources for a dataset, for each specific cloud provider.
+
+Configuration
+-------------
+
+The CPG configures our infrastructure in a private repository, which is structured like:
+
+- ``config-infrastructure.yaml``:
+    - Contains general configuration with org constants and cloud provider details.
+- ``config-datasets.yaml``:
+    - Contains dataset specific configuration, like the dataset name, and the cloud environments to use for each dataset.
+    - This matches the config.
+
+    .. code-block:: yaml
+
+       dataset:
+         # other datasets this one need to READ but not write
+         depends_on_readonly:
+           - dataset-1
+           - dataset-2
+         # other datasets this one need to READ + WRITE
+         depends_on:
+           - dataset-3
+         enable_release: true
+         enable_shared_project: true
+         # read the user.rst docs on these params
+         is_internal_dataset: true
+         gcp:
+           project: dataset-gcp-project-id
+
+- ``users.yaml``
+    - Contains the users and their roles in the CPG infrastructure.
+    - This is used to manage group memberships.
+
+    .. code-block:: yaml
+
+       firstname.lastname:
+       can_access_internal_dataset_logs: true
+       gcp:
+         hail_batch_username: hailbatchusername
+         # an email recognised by google identity, to add to google groups
+         id: firstname.lastname@populationgenomics.org.au
+
+- ``projects.yaml``
+    - A list of legal projects / protocols that we use to double check people have access to the datasets. This isn't used in the infra code directly, it's a process check on top which blocks CI merges.
```

Comment:
A bit of a nitpick, but this is technically used in the infra code to create the list of projects on a user: https://github.com/populationgenomics/cpg-infrastructure/blob/main/cpg_infra/config/config.py#L37
I don't think that list is actually used anywhere though.

Reply:
Yeah, I think we should just remove it, probably meant to be something more that never eventuated.